src/xenia/app/xenia_main.cc

```diff
@@ -39,26 +40,31 @@
 
 #include "third_party/xbyak/xbyak/xbyak_util.h"
 
-DEFINE_string(apu, "any", "Audio system. Use: [any, nop, xaudio2]");
-DEFINE_string(gpu, "any", "Graphics system. Use: [any, vulkan, null]");
-DEFINE_string(hid, "any", "Input system. Use: [any, nop, winkey, xinput]");
+CVar apu_("AudioSystem", "any", "Audio system. Use: [any, nop, xaudio2]", "General");
```

Comment:
I suggest maybe [using a singleton](https://github.com/Wildenhaus/xenia/blob/qt-experimental/src/xenia/app/library/game_library.cc#L8) or some other global system for managing configuration variables and and registering callbacks (like Dr. Chat suggested earlier and you have already started towards implementing). That way it's self contained, can be referenced from anywhere, and you don't need to do checks/handling in `xenia_main.cc`. Those can go in their component-specific files.

Reply:
Ugh, looks like we're going to confuse jgoyvaerts even more with contradicting suggestions :D But the choice depends primarily on whether some initialization code is required for the whole system, I think. Not very important probably though.

Reply:
I overlooked the part where you're using a global map to store CVars, which is essentially what I was suggesting. Carry on 😛